---
title: "main_figures"
author: "Jared Sipes"
date: "2025-02-11"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```




# Summary 

The purpose of this R notebook is to create the figures for the paper:

"Spatial Transcriptomic Profiling of the Human Fallopian Tube Epithelium Reveals Region-specific Gene Expression Patterns"

This document uses a GeoMx dataset that has already undergone Quality control and normalization. 

For details, see: geomx_dataset_and_normalization.Rmd



# 0 Setup

# 0.1 Load Packages
 
The following packages are needed for this document. 


```{r}
# 
# if (!require("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")

# The following initializes most up to date version of Bioc
# BiocManager::install(version="3.15")
# 
# BiocManager::install("NanoStringNCTools")
# BiocManager::install("GeomxTools")
# BiocManager::install("GeoMxWorkflows")

# Note:
# Needed to install package lme4, numderiv
library(NanoStringNCTools)
library(GeomxTools)
library(GeoMxWorkflows)

if(packageVersion("GeomxTools") < "2.1" & 
   packageVersion("GeoMxWorkflows") >= "1.0.1"){
    stop("GeomxTools and Workflow versions do not match. Please use the same version. 
    This workflow is meant to be used with most current version of packages. 
    If you are using an older version of Bioconductor please reinstall GeoMxWorkflows and use vignette(GeoMxWorkflows) instead")
}

if(packageVersion("GeomxTools") > "2.1" & 
   packageVersion("GeoMxWorkflows") <= "1.0.1"){
    stop("GeomxTools and Workflow versions do not match. 
         Please use the same version, see install instructions above.")
    
    # to remove current package version
        # remove.packages("GeomxTools")
        # remove.packages("GeoMxWorkflows")
    # see install instructions above 
}
```

```{r}
# for file names
library(here)

# read in xl files 
library(readxl)


# Need for UMAP, tSNE plots
library(umap)
library(Rtsne)

# Needed for the heatmap plotting
library(ComplexHeatmap)  #BiocManager::install("ComplexHeatmap")

#Needed for volcano plotting functions

library(ggrepel)
library(ggplot2)
library(latex2exp)
library(plyr)
library(dplyr)


# Read in images and convert to ggplot object

library(magick)


# used for some color palettes  
library(RColorBrewer)
library(stringi)


# For significance
library(ggpubr)


# For Stitching multiple plots together
library(patchwork)
library(cowplot)
library(grid)

library(tidyr)

```


For space purposes, some functions are moved to an outside document "functions.R".


```{r, include=FALSE}
source("functions.R", local = knitr::knit_global())
# or sys.source("your-script.R", envir = knitr::knit_global())
```

# 0.2 Load the Dataset and all Analyses Used

```{r}

# make sure to use correct "here" function.

load(here::here("all_q_norm.Rdata"))

load(here::here("disc_q_norm.Rdata"))

load(here::here("valid_q_norm.Rdata"))


dim(all_q_norm)

dim(disc_q_norm)

dim(valid_q_norm)

```




Refactor the GeoMx objects to use simplified region names. 

```{r}

factor_region <- function(GeoMxObj = all_q_norm){
  pData(GeoMxObj)$region <- factor(pData(GeoMxObj)$region, levels = c("Fimbria", "Infundibulum", "Ampulla", "Isthmus"))
  
  pData(GeoMxObj)$region <- revalue(pData(GeoMxObj)$region, c("Fimbria"="Fim", "Infundibulum"="Inf", "Ampulla" = "Amp", "Isthmus" = "Isth"))
  
  return(GeoMxObj)
}


all_q_norm <- all_q_norm |> factor_region()

disc_q_norm <- disc_q_norm |> factor_region()

valid_q_norm <- valid_q_norm |> factor_region()

```



Subset the "all" dataset into discovery and validation datasets.

The major difference between these datasets and the "disc_q_norm" and "valid_q_norm" is that the latter were normalizzed separately. In contrast, all_disc and all_valid were normalized together and comparisons of gene expression between the two datasets are valid. 

```{r}
all_disc <- all_q_norm[, pData(all_q_norm)$Patient %in% c("P1", "P2", "P3")]

`%ni%` <- Negate(`%in%`)

all_valid <- all_q_norm[, pData(all_q_norm)$Patient %ni% c("P1", "P2", "P3")]


```


# 0.3 Load Stat Analyses


### Load files required for volcano plots


```{r}


stat_files <- paste0(getwd(), "/stat_files")

stat_files_all <- paste0(stat_files, "/ALL")
stat_files_disc <- paste0(stat_files, "/Discovery")
stat_files_valid <- paste0(stat_files, "/Validation")

#secretory cells
pairwise_s_disc <- paste0(stat_files_disc, "/0.10_SECRETORY_Pairwise_region_comparison.xlsx")
pairwise_s_valid <- paste0(stat_files_valid, "/Validation_SECRETORY_Pairwise_region_comparison.xlsx")
pairwise_s_all <- paste0(stat_files_all, "/ALL_SECRETORY_Pairwise_region_comparison.xlsx")
# ciliated cells 
pairwise_c_disc <- paste0(stat_files_disc, "/0.10_CILIATED_Pairwise_region_comparison.xlsx")
pairwise_c_valid <- paste0(stat_files_valid, "/Validation_CILIATED_Pairwise_region_comparison.xlsx")
pairwise_c_all <- paste0(stat_files_all, "/ALL_CILIATED_Pairwise_region_comparison.xlsx")

# create a nested list structure to store all of the dataframes in one object

pairwise_comparisons <- list()

```




#### FUNCT: get_all_comparisons


```{r}

comparison_pairwise <- c("Fimbria.vs.Infundibulum", "Fimbria.vs.Ampulla", "Fimbria.vs.Isthmus", 
                     "Infundibulum.vs.Ampulla", "Infundibulum.vs.Isthmus", "Ampulla.vs.Isthmus")

pairwise_names <- c("fim_v_inf", "fim_v_amp", "fim_v_isth", "inf_v_amp", "inf_v_isth", "amp_v_isth")

get_all_comparions <- function(path = pairwise_s_disc, comparison_list = comparison_pairwise, comparison_name = pairwise_names){
  data_list <- lapply(comparison_list, read_xlsx, path = path)
  names(data_list) <- comparison_name
  
  return(data_list)
}


```


Create list of pairwise comparisons

```{r}

pairwise_comparisons[["disc"]][["sec"]] <- get_all_comparions(path = pairwise_s_disc)
pairwise_comparisons[["valid"]][["sec"]] <- get_all_comparions(path = pairwise_s_valid)
pairwise_comparisons[["all"]][["sec"]] <- get_all_comparions(path = pairwise_s_all)

pairwise_comparisons[["disc"]][["cil"]] <- get_all_comparions(path = pairwise_c_disc)
pairwise_comparisons[["valid"]][["cil"]] <- get_all_comparions(path = pairwise_c_valid)
pairwise_comparisons[["all"]][["cil"]] <- get_all_comparions(path = pairwise_c_all)

```


### 2.2 Loading One vs Other Comparisons

In this comparison, one region is comared to the average of all other regions, for four total comparisons

1. fimbria v others
2. Inf v others
3. amp v others
4. isth v others


Excel doc locations

```{r}

one_v_other_s_disc <- paste0(stat_files_disc, "/0.10_SECRETORY_One.VS.Others_region_comparison.xlsx")
one_v_other_s_valid <- paste0(stat_files_valid, "/Validation_SECRETORY_One.VS.Others_region_comparison.xlsx")
one_v_other_s_all <- paste0(stat_files_all, "/ALL_SECRETORY_One.VS.Others_region_comparison.xlsx")

one_v_other_c_disc <- paste0(stat_files_disc, "/0.10_CILIATED_One.VS.Others_region_comparison.xlsx")
one_v_other_c_valid <- paste0(stat_files_valid, "/Validation_CILIATED_One.VS.Others_region_comparison.xlsx")
one_v_other_c_all <- paste0(stat_files_all, "/ALL_CILIATED_One.VS.Others_region_comparison.xlsx")


```

```{r}

one_v_other <- list()


comparison_one_v_other <- c("Fimbria.vs.Others", "Infundibulum.vs.Others", "Ampulla.vs.Others", "Isthmus.vs.Others")

one_v_other_names <- c("fim_v_other", "inf_v_other", "amp_v_other", "isth_v_other")

one_v_other[["disc"]][["sec"]] <- get_all_comparions(path = one_v_other_s_disc, comparison_list = comparison_one_v_other, comparison_name = one_v_other_names)
one_v_other[["valid"]][["sec"]] <- get_all_comparions(path = one_v_other_s_valid, comparison_list = comparison_one_v_other, comparison_name = one_v_other_names)
one_v_other[["all"]][["sec"]] <- get_all_comparions(path = one_v_other_s_all, comparison_list = comparison_one_v_other, comparison_name = one_v_other_names)


one_v_other[["disc"]][["cil"]] <- get_all_comparions(path = one_v_other_c_disc, comparison_list = comparison_one_v_other, comparison_name = one_v_other_names)
one_v_other[["valid"]][["cil"]] <- get_all_comparions(path = one_v_other_c_valid, comparison_list = comparison_one_v_other, comparison_name = one_v_other_names)
one_v_other[["all"]][["cil"]] <- get_all_comparions(path = one_v_other_c_all, comparison_list = comparison_one_v_other, comparison_name = one_v_other_names)

```


# F1. Figure 1

Figure 1A-F are images and are not found in this document. 

Figure 1G was generated in the data_analysis document.



## F1.2 Figure 1H 



```{r  fig.height=6, fig.width=15}


text_size <- 25

# For Discovery Dataset

FOXJ1_d <- cil_v_sec_boxplot_points(gene = "FOXJ1", dataset = disc_q_norm) + theme(text=element_text(size=text_size)) + theme(legend.position = "none")

PAX8_d <- cil_v_sec_boxplot_points(gene = "PAX8", dataset = disc_q_norm) + theme(text=element_text(size=text_size)) + theme(legend.position = "none")


# For Validation Dataset

FOXJ1_v <- cil_v_sec_boxplot_points(gene = "FOXJ1", dataset = valid_q_norm) + theme(text=element_text(size=text_size)) + theme(legend.position = "none")

PAX8_v <- cil_v_sec_boxplot_points(gene = "PAX8", dataset = valid_q_norm) + theme(text=element_text(size=text_size)) + theme(legend.position = "none")



title1 <- grid::textGrob(label = "Discovery", gp=gpar(fontsize=35))
title2 <- grid::textGrob(label = "Validation", gp=gpar(fontsize=35))

# Patchwork is used to combine plots

combine_1 =  (wrap_elements(panel = title1) / (FOXJ1_d + PAX8_d)) + plot_layout(heights = c(1.5, 10))
combine_2 =  (wrap_elements(panel = title2) / (FOXJ1_v + PAX8_v)) + plot_layout(heights = c(1.5, 10))


wrap_elements(combine_1)|wrap_elements(combine_2)
```




# F2. Figure 2


### F2.A-D Secretory Discovery v Validation

The volcano plots used in Figure 2 are generated below.

Some slight differences in genes labeled are expected between different generations. 

```{r, Secretory Segmements Comparison, fig.height=10, fig.width=20}
genes_fim_v_other_sec <- get_common_genes(df_1 = one_v_other$disc$sec$fim_v_other, df_2 = one_v_other$valid$sec$fim_v_other, type = "both", FC_cutoff = 0.2)

genes_inf_v_other_sec <- get_common_genes(df_1 = one_v_other$disc$sec$inf_v_other, df_2 = one_v_other$valid$sec$inf_v_other, type = "both", FC_cutoff = 0.2)

genes_amp_v_other_sec <- get_common_genes(df_1 = one_v_other$disc$sec$amp_v_other, df_2 = one_v_other$valid$sec$amp_v_other, type = "both", FC_cutoff = 0.2)

genes_isth_v_other_sec <- get_common_genes(df_1 = one_v_other$disc$sec$isth_v_other, df_2 = one_v_other$valid$sec$isth_v_other, type = "both", FC_cutoff = 0.2)


title = 14

disc_sec <- one_v_other$disc$sec
a <- volcano_plot(df = disc_sec$fim_v_other, 
                 title_name = "Fimbria Secretory Cells, Discovery", title_size = title, upreg = "Fimbria", downreg = "Others",
                 #highlight_genes = genes_fim_v_other_sec
                 ) |> add_labels(df = disc_sec$fim_v_other)
  
b <- volcano_plot(df = disc_sec$inf_v_other, 
                 title_name = "Infundibulum Secretory Cells, Discovery", title_size = title, upreg = "Infundibulum", downreg = "Others",
                 #highlight_genes = genes_inf_v_other_sec
                 ) |> add_labels(df = disc_sec$inf_v_other) |> remove_legend()

c <- volcano_plot(df = disc_sec$amp_v_other, 
                 title_name = "Ampulla Secretory Cells, Discovery", title_size = title, upreg = "Ampulla", downreg = "Others",
                 #highlight_genes = genes_amp_v_other_sec
                 ) |> add_labels(df = disc_sec$amp_v_other) |> remove_legend()

d <- volcano_plot(df = disc_sec$isth_v_other, 
                 title_name = "Isthmus Secretory Cells, Discovery", title_size = title, upreg = "Isthmus", downreg = "Others",
                 #highlight_genes = genes_isth_v_other_sec
                 ) |> add_labels(df = disc_sec$isth_v_other) |> remove_legend()


valid_sec <- one_v_other$valid$sec

w <- volcano_plot(df = valid_sec$fim_v_other, 
                 title_name = "Fimbria Secretory Cells, Validation", title_size = title, upreg = "Fimbria", downreg = "Others",
                 #highlight_genes = genes_fim_v_other_sec
                 ) |> add_labels(df = valid_sec$fim_v_other) |> remove_legend()
  
x <- volcano_plot(df = valid_sec$inf_v_other, 
                 title_name = "Infundibulum Secretory Cells, Validation", title_size = title, upreg = "Infundibulum", downreg = "Others",
                 #highlight_genes = genes_inf_v_other_sec
                 ) |> add_labels(df = valid_sec$inf_v_other) |> remove_legend()

y <- volcano_plot(df = valid_sec$amp_v_other, 
                 title_name = "Ampulla Secretory Cells, Validation", title_size = title, upreg = "Ampulla", downreg = "Others",
                 #highlight_genes = genes_amp_v_other_sec
                 ) |> add_labels(df = valid_sec$amp_v_other) |> remove_legend()

z <- volcano_plot(df = valid_sec$isth_v_other, 
                 title_name = "Isthmus Secretory Cells, Validation", title_size = title, upreg = "Isthmus", downreg = "Others",
                 #highlight_genes = genes_isth_v_other_sec
                 ) |> add_labels(df = valid_sec$isth_v_other) |> remove_legend()

top <-   grobTree(rectGrob(gp=gpar(fill='#F0F0F0',col= 'black')), textGrob('Discovery Dataset, Secretory Cells', gp=gpar(fontsize=25)))

bottom <- grobTree(rectGrob(gp=gpar(fill='#F0F0F0',col= 'black')), textGrob('Validation Dataset, Secretory Cells', gp=gpar(fontsize=25)))

wrap_elements(top) /(a | b | c | d) / wrap_elements(bottom) /(w | x | y | z)  + plot_layout(heights = c(1,5,1,5), guides = "collect") + plot_annotation(tag_levels = list(c("", "a", "b", "c", "d", "", "e", "f", "g", "h"))) & 
  theme(plot.tag = element_text(size = 26))


```

### F2.G-J Ciliated Discovery v Validation

```{r, Ciliated Segmements Comparison, fig.height=10, fig.width=20}


genes_fim_v_other_sec <- get_common_genes(df_1 = one_v_other$disc$cil$fim_v_other, df_2 = one_v_other$valid$cil$fim_v_other, type = "both", FC_cutoff = 0.2)

genes_inf_v_other_sec <- get_common_genes(df_1 = one_v_other$disc$cil$inf_v_other, df_2 = one_v_other$valid$cil$inf_v_other, type = "both", FC_cutoff = 0.2)

genes_amp_v_other_sec <- get_common_genes(df_1 = one_v_other$disc$cil$amp_v_other, df_2 = one_v_other$valid$cil$amp_v_other, type = "both", FC_cutoff = 0.2)

genes_isth_v_other_sec <- get_common_genes(df_1 = one_v_other$disc$cil$isth_v_other, df_2 = one_v_other$valid$cil$isth_v_other, type = "both", FC_cutoff = 0.2)


title = 14

disc_cil <- one_v_other$disc$cil
a <- volcano_plot(df = disc_cil$fim_v_other, 
                 title_name = "Fimbria Ciliated Cells, Discovery", title_size = title, upreg = "Fimbria", downreg = "Others",
                 #highlight_genes = genes_fim_v_other_sec
                 ) |> add_labels(df = disc_cil$fim_v_other)
  
b <- volcano_plot(df = disc_cil$inf_v_other, 
                 title_name = "Infundibulum Ciliated Cells, Discovery", title_size = title, upreg = "Infundibulum", downreg = "Others",
                 #highlight_genes = genes_inf_v_other_sec
                 ) |> add_labels(df = disc_cil$inf_v_other) |> remove_legend()

c <- volcano_plot(df = disc_cil$amp_v_other, 
                 title_name = "Ampulla Ciliated Cells, Discovery", title_size = title, upreg = "Ampulla", downreg = "Others",
                 #highlight_genes = genes_amp_v_other_sec
                 ) |> add_labels(df = disc_cil$amp_v_other) |> remove_legend()

d <- volcano_plot(df = disc_cil$isth_v_other, 
                 title_name = "Isthmus Ciliated Cells, Discovery", title_size = title, upreg = "Isthmus", downreg = "Others",
                 #highlight_genes = genes_isth_v_other_sec
                 ) |> add_labels(df = disc_cil$isth_v_other) |> remove_legend()


valid_cil <- one_v_other$valid$cil

w <- volcano_plot(df = valid_cil$fim_v_other, 
                 title_name = "Fimbria Ciliated Cells, Validation", title_size = title, upreg = "Fimbria", downreg = "Others",
                 #highlight_genes = genes_fim_v_other_sec
                 ) |> add_labels(df = valid_cil$fim_v_other) |> remove_legend()
  
x <- volcano_plot(df = valid_cil$inf_v_other, 
                 title_name = "Infundibulum Ciliated Cells, Validation", title_size = title, upreg = "Infundibulum", downreg = "Others",
                 #highlight_genes = genes_inf_v_other_sec
                 ) |> add_labels(df = valid_cil$inf_v_other) |> remove_legend()

y <- volcano_plot(df = valid_cil$amp_v_other, 
                 title_name = "Ampulla Ciliated Cells, Validation", title_size = title, upreg = "Ampulla", downreg = "Others",
                 #highlight_genes = genes_amp_v_other_sec
                 ) |> add_labels(df = valid_cil$amp_v_other) |> remove_legend()

z <- volcano_plot(df = valid_cil$isth_v_other, 
                 title_name = "Isthmus Ciliated Cells, Validation", title_size = title, upreg = "Isthmus", downreg = "Others",
                 #highlight_genes = genes_isth_v_other_sec
                 ) |> add_labels(df = valid_cil$isth_v_other) |> remove_legend()

top <-   grobTree(rectGrob(gp=gpar(fill='#F0F0F0',col= 'black')), textGrob('Discovery Dataset, Ciliated Cells', gp=gpar(fontsize=25)))

bottom <- grobTree(rectGrob(gp=gpar(fill='#F0F0F0',col= 'black')), textGrob('Validation Dataset, Ciliated Cells', gp=gpar(fontsize=25)))

wrap_elements(top) /(a | b | c | d) / wrap_elements(bottom) /(w | x | y | z)  + plot_layout(heights = c(1,5,1,5), guides = "collect") + plot_annotation(tag_levels = list(c("", "a", "b", "c", "d", "", "e", "f", "g", "h"))) & 
  theme(plot.tag = element_text(size = 26))

```

# F3. Figure 3 


## F3.1 Figure 3A


Figure 3A is an image generated using BioRender. It is read in below and converted to a ggplot object. 


```{r}

# image_ggplot is from the package "magick"

picture_3A <- image_read(paste0(getwd(), "/pictures/fig3a.jpeg")) |>
  image_ggplot()

```



## F3.2 Figure 3B-C (Volcano plots with mature ciliated cell markers)



#### Markers of Mature Ciliated Cells 


The following markers of mature ciliated cells are identified using single cell sequencing by Ulrich et al. 

Cellular heterogeneity of human fallopian tubes in normal and hydrosalpinx disease states identified using scRNA-seq

DOI: 10.1016/j.devcel.2022.02.017


```{r}

mature_cil <- c(
  "CAPS", "CETN2", "PIFO", "C1orf194", "C9orf24", "SNTN", "AGR3", "C20orf85", "MORN2", "PERP", "HMGN3", "TMEM59", "RSPH1",
  "CALM1", "AGR2", "METRN", "DYNLL1", "CD24", "TXN", "PRDX5", "C9orf116", "IK", "MORN5", "FAM183A", "CFAP126", "CAPSL",
  "IGFBP7", "C11orf88", "SPA17", "ODF3B", "CCDC146", "DYNLRB2", "FAM92B", "C9orf135", "PSENEN", "C5orf49", "FAM81B", "ARHGAP18", "PPIL6",
  "SLC44A4", "RBP1", "ANKRD66", "CYSTM1", "MLF1", "CFAP52", "WDR54", "LRRC23", "CTSS", "TAX1BP1", "TPPP3", "GON7", "FAM174A",
  "C6", "TSPAN3", "CIB1", "EFCAB1", "IFT57", "CCDC170", "DYDC2", "ARMC3", "NME5", "ENKUR", "MNS1", "TCTN1", "SMIM22",
  "C11orf97", "WDR78", "MS4A8", "TSPAN6", "FAM216B", "RIIAD1", "FOXJ1", "ARL3", "DRC1", "CTXN1", "ROPN1L", "NUCB2", "EZR",
  "GSTA3", "C4orf48", "MT-ND3", "NELL2", "CCDC113", "SPAG6", "AC013264.1", "GET1", "FXYD3", "TFF3", "EFCAB10", "STK33", "SPATA17",
  "FAIM", "CABCOCO1", "CKB", "TTC29", "KIF21A", "SCGB2A1", "AL357093.2", "STOML3", "CFAP53", "WDR86-AS1", "DNAH9", "C20orf96", "TUBA1A",
  "LDLRAD1", "SAMHD1", "SOD3", "DNAH5", "CCDC173", "ERICH3", "CTSD", "MT-ND5", "TSPAN1", "PLTP", "AK1", "FOLR1", "RHOB",
  "FABP6", "CST6", "PDK4", "ANXA13", "RHOU", "IRX3", "C12orf75", "NEDD9", "MOSPD1", "TMEM190", "CRIP1", "MT-ND4L", "GOLM1",
  "ASRGL1", "SCGB1D4", "GSTA1", "PGR", "G0S2", "FGA", "RRAD"
)


```






### Volc_plot Fimbria v Isthmus Comparisons


```{r}
title = 20

df1 <- pairwise_comparisons$disc$cil

df2 <- pairwise_comparisons$valid$cil

fim_v_isth_disc <- volcano_plot(df = df1$fim_v_isth, 
                 title_name = "Fimbria v Isthmus, Ciliated Cells, Discovery", title_size = title, upreg = "Fimbria", downreg = "Isthmus",
                 highlight_genes = mature_cil
                 ) |> add_labels(df = df1$fim_v_isth, lab_size = 5, highlight_genes = mature_cil, n_genes =0) |> remove_legend()


fim_v_isth_valid <- volcano_plot(df = df2$fim_v_isth, 
                 title_name = "Fimbria v Isthmus, Ciliated Cells, Validation", 
                 title_size = title, upreg = "Fimbria", downreg = "Isthmus",
                 highlight_genes = mature_cil
                 ) |> add_labels(df = df2$fim_v_isth, lab_size = 5, highlight_genes = mature_cil, n_genes =0) |> remove_legend()



# volcano_plot(df = df2$fim_v_isth, 
#                  title_name = "Fimbria v Isthmus, Ciliated Cells, Validation", 
#                  title_size = title, upreg = "Fimbria", downreg = "Isthmus",
#                  highlight_genes = mature_cil
#                  )


```





## F3.3 Create Ratio Comparisons

The following functions create ratio comparison plots for figure 3D-E (discovery) and 3J-K (validation).


### FUNCT: make_ratio()

```{r}



# Ascending Secretory approaching Isthmus



make_ratio <- function(gene1 = "PAX8", gene2 = "FOXJ1", dataset = all_q_norm, title = "All", point_size = 3, selected_segment = NA, make_legend = F){
  
    dataset_original <- dataset
    
    
    # Filter dataset for the selected segment if provided
    if (!is.na(selected_segment)) {
        dataset <- dataset[, pData(dataset)$segment == selected_segment]
    }
  
    # Compute gene1/gene2 ratio and convert to vector

    vector <- as.vector(assayDataElement(dataset[gene1, ], elt = "q_norm")/ 
                          assayDataElement(dataset[gene2, ], elt = "q_norm"))
    
    # Create the ggplot
    plot <- ggplot(
      pData(dataset), 
      aes(x = region,
          y = vector,
          fill = region
          )
      )+
      geom_boxplot(outlier.shape = NA) +
      labs(title = TeX(paste0("\\textit{", gene1, "/", gene2, "}")), 
           y = TeX(paste0("\\textit{", gene1, "/", gene2, "}")),
           x = "Anatomical Region")+
      facet_wrap(~segment)+
  
      geom_jitter(size = point_size, width = 0.25, stroke = 1.3, aes(shape = Patient))+
  
      scale_shape_manual(values = c(3, 4, 13, 21, 22, 23, 24))+
  
      theme_bw()+
  
      # scale_x_discrete(labels=c("Fim","Inf","Amp", "Isth"))+
      scale_fill_manual(values= c("#7CAE00", "#00C1C6", "#F8766D", "#C77CFF"))+
  
      theme(axis.text.x = element_text(size = 15),
            axis.text.y = element_text(size = 12),
            axis.title.y = element_text(size = 15),
            axis.title.x = element_text(size = 15),
            legend.text = element_text(size =12),
            legend.title = element_text(size = 12),
            strip.text = element_text(size = 15))


  # Generate legend and adjust plot sizes
  legend <- make_boxplot_legend(dataset = dataset_original)
  
  ratio_resize <- adj_text_size(plot, pnt = 17)
  ratio_remove_legend <- remove_legend(ratio_resize)
  ratio_sig <- add_sig(ratio_remove_legend, pnt = 4)

  # Combine the main plot and legend
  
  if(make_legend){
  plot_final <- (ratio_sig) / legend + plot_layout(heights = c(5, 1))
  
  return(plot_final)
  }
  plot_final <- ratio_sig
  
  return(plot_final)
  
}

```


### Create Ratio Comparisons

```{r Ratio Comparison, fig.height=5, fig.width=10}


# make_ratio(dataset = all_q_norm, title = "All")


# PAX8 / FOXJ1 plots

PvF_disc <- make_ratio(gene1 = "PAX8", gene2 = "FOXJ1", dataset = disc_q_norm, selected_segment = "Secretory") 


PvF_valid <- make_ratio(gene1 = "PAX8", gene2 = "FOXJ1", dataset = valid_q_norm, selected_segment = "Secretory") 


# FOXJ1/PAX8 plots


FvP_disc <- make_ratio(gene1 = "FOXJ1", gene2 = "PAX8", dataset = disc_q_norm, selected_segment = "Ciliated") 


FvP_valid <- make_ratio(gene1 = "FOXJ1", gene2 = "PAX8", dataset = valid_q_norm, selected_segment = "Ciliated") 




```



## F3.4 Create Boxplot comparisons

The following sections create the other boxplots for figure 3.

Discovery boxplots Fig3F-I
and 
Validation boxplots Fig3L-O




### FUNCT: make_fig3()

This create boxplots formatted specifically for figure 3


```{r}

# List of Genes up in Mature Ciliated Cells 

Fig3 <- c("MLF1", "SPA17", "CTSS", "C6")


# Function to create boxplots for Figure 3

make_fig3 <- function(genes = Fig3, dataset = all_q_norm, title = "All"){

  graphs_up_Fig3 <- lapply(X = genes, FUN = Cil_v_Sec_Anat_Boxplot_Points, dataset = dataset)
  
  
  
  Fig3_resize <- lapply(graphs_up_Fig3, adj_text_size, pnt = 17)
  Fig3_remove_legend <- lapply(Fig3_resize, remove_legend)
  Fig3_sig <- lapply(Fig3_remove_legend, add_sig, pnt = 4)



  return(Fig3_sig)
  
  }


```





### Generate Boxplots

```{r}

F3_disc <- make_fig3(genes = Fig3, dataset = disc_q_norm)

F3_valid <- make_fig3(genes = Fig3, dataset = valid_q_norm)

legend_disc <- make_boxplot_legend(dataset = disc_q_norm)

legend_valid <- make_boxplot_legend(dataset = valid_q_norm)

```



## F3.5 Generate Final Image


```{r fig.height=20, fig.width=25}


library(patchwork)



# Generate Labels for adding to the graphs

top <- grid::grobTree(rectGrob(gp=gpar(fill='#F0F0F0',col= 'black')), textGrob('Discovery Dataset', gp=gpar(fontsize=25))) 

bottom <- grid::grobTree(rectGrob(gp=gpar(fill='#F0F0F0',col= 'black')), textGrob('Validation Dataset', gp=gpar(fontsize=25))) 



# Use patchwork to combine the final image into one complete image

wrap_elements(free(picture_3A) | fim_v_isth_disc | fim_v_isth_valid) / 
  wrap_elements(top) /

 (((FvP_disc| PvF_disc ) | F3_disc[[1]] | F3_disc[[2]] |  F3_disc[[3]] | F3_disc[[4]]) +
    plot_layout(widths = c(1, 1, 2, 2, 2, 2)))/

 legend_disc /

 wrap_elements(bottom) /

 (((FvP_valid| PvF_valid) | F3_valid[[1]] | F3_valid[[2]] | F3_valid[[3]] |F3_valid[[4]]) +
    plot_layout(widths = c(1, 1, 2, 2, 2, 2)))/

 legend_valid +

 plot_layout(heights = c(5, 0.5, 4, 0.5, 0.5, 4, 0.5))



```




# F4. Figure 4: Adhesion, ROS and Apoptosis Related



## FUNCT: make_fig4()

This create boxplots formatted for figure 4. 


```{r}

# This function makes all of the boxplots for figure 4 according to the same specifications

make_fig4 <- function(genes = Fig4_gene_list, dataset = all_q_norm, title = "All"){

  graphs_up_Fig <- lapply(X = genes, FUN = Cil_v_Sec_Anat_Boxplot_Points, dataset = dataset)
  
  Fig_resize <- lapply(graphs_up_Fig, adj_text_size, pnt = 17)
  Fig_remove_legend <- lapply(Fig_resize, remove_legend)
  Fig_sig <- lapply(Fig_remove_legend, add_sig, pnt = 4)

  return(Fig_sig)
  
  }
```


## Create fig4 boxplots

```{r Make all Gene Comparisons, fig.height=22, fig.width=20}

# List of 12 genes to be included in the Figure 4 Boxplots

Fig4_gene_list <- c("ESR1", "PGR", "AR", "FOXA2", "CDH1", "CDH3", "CD99", "LGALS3", "PRDX5", "TXNIP", "GAS1", "BAD", "FOXA1", "FOXA2", "PBX1", "ANXA1", "WT1", "BIRC3", "TNFSF10")

  

# Create all of the Boxplots for the Discovery Dataset

F4_disc <- make_fig4(genes = Fig4_gene_list , dataset = disc_q_norm)
names(F4_disc) <- Fig4_gene_list

# Create all of the Boxplots for the Validation Dataset

F4_valid <- make_fig4(genes = Fig4_gene_list , dataset = valid_q_norm)
names(F4_valid) <- Fig4_gene_list


# Create Grobs for use as labels

top_1 <-   grobTree(rectGrob(gp=gpar(fill='#F0F0F0',col= 'black')), textGrob('Discovery Dataset: Cell-Cell Adhesion Transcripts', gp=gpar(fontsize=25)))

bottom_1 <- grobTree(rectGrob(gp=gpar(fill='#F0F0F0',col= 'black')), textGrob('Validation Dataset: Cell-Cell Adhesion Transcripts', gp=gpar(fontsize=25)))


top_2 <-   grobTree(rectGrob(gp=gpar(fill='#F0F0F0',col= 'black')), textGrob('Discovery Dataset: ROS and Apoptosis Related Transcripts', gp=gpar(fontsize=25)))

bottom_2 <- grobTree(rectGrob(gp=gpar(fill='#F0F0F0',col= 'black')), textGrob('Validation Dataset: ROS and Apoptosis Related Transcripts', gp=gpar(fontsize=25)))


  # Part 1 - Cell Cell Adhesion
  wrap_elements(top_1) / 
  (F4_disc$CDH1 | F4_disc$CDH3 |  F4_disc$CD99 | F4_disc$LGALS3) /
  legend_disc /
  wrap_elements(bottom_1) / 
  (F4_valid$CDH1 | F4_valid$CDH3 |  F4_valid$CD99 | F4_valid$LGALS3) /
  legend_valid /
  # Part 2 - ROS and Apoptosis related  
  wrap_elements(top_2) / 
  (F4_disc$PRDX5 | F4_disc$TXNIP |  F4_disc$BAD | F4_disc$GAS1) /
  legend_disc /
  wrap_elements(bottom_2) / 
  (F4_valid$PRDX5 | F4_valid$TXNIP |  F4_valid$BAD | F4_valid$GAS1) /
  legend_valid +
    
    plot_layout(heights = c(0.5, 4, 0.5, 0.5, 4, 0.5, 0.5, 4, 0.5, 0.5, 4, 0.5))  
    
    # plot_annotation(tag_levels = list(c("", "a", "b", "c", "d", 
    #                                     "", "", "e", "f", "g", "h", 
    #                                     "", "","i", "j", "k", "l", 
    #                                     "", "", "m", "n", "o", "p", "")), 
    #                                theme = theme(plot.tag = element_text(size = 30, face = "bold")))





```




# F5. Figure 5: MHC-II related transcripts


## Load OVGP1 Data

```{r}

#read in from excel
OVGP1_percent <- read_xlsx(path = paste0(getwd(), "/IHC_analysis/OVGP1 Percentages.xlsx"))

# adjust to tidy data format
OVGP1 <- OVGP1_percent |> pivot_longer(cols = c("Fim", "Inf", "Amp", "Isth"), names_to = "region", values_to = "percent") |> mutate(percent = as.numeric(percent))

# factor region 
OVGP1$region <- OVGP1$region |> factor(levels = c("Fim", "Inf", "Amp", "Isth"))

# remove slides not in GeoMx dataset 
OVGP1_subset <- OVGP1[OVGP1$patient_ID %in% unique(pData(all_q_norm)$patient_ID),]


# this is necessary to merge with the GeoMx Object. 
OVGP1_subset <- transform(OVGP1_subset, patient_ID = as.double(patient_ID))


# merge OVGP1 data
pData(all_q_norm) <- pData(all_q_norm) |> left_join(OVGP1_subset |> select(region, patient_ID, percent), by = c("region", "patient_ID"))

```





### FUNCT: compare_OVGP1()

```{r}

compare_OVGP1 <- function(dataset = all_q_norm, gene = "HLA-DRA", title_append = "Expression Inversely Correlates with Percentage OVGP1+ Cells"){
  # get expression of selected gene 
  
  gene_vector <- as.vector(assayDataElement(dataset, elt = "q_norm")[gene, ])

  plot <- ggplot(data = pData(dataset), aes(x = percent, y = gene_vector))+
    geom_point(aes(color = Patient))+
    labs(title = TeX(paste0("\\textit{", gene, "} ", title_append)), 
         y = TeX(paste0("\\textit{", gene, "} ", "Expression (GeoMx)")),
         x = "Percent OVGP1+ Cells in Region (IHC staining)")+
    geom_smooth(formula = y ~ x, method = "lm")+ 
    stat_cor(method="pearson", size = 5)+
    theme_classic()

  return(plot)
}

```



### FUNCT: calc_pearson(), calc_pearson_sig()

```{r}

# estimate the value of the pearson correlation of all genes compared to the selected gene

calc_pearson <- function(dataset = all_q_norm, gene ="ESR1"){
  
  # get expression data for the 
  
  expression_data <- assayDataElement(dataset, elt = "q_norm")
  
  # get expression data for just the gene of interest
  target_expression <- assayDataElement(dataset, elt = "q_norm")[gene, ]
  
  correlations <- apply(expression_data, 1, 
                        function(gene_expression){
                          # function to compare correlation between gene and target selected
                          cor(gene_expression, target_expression, method = "pearson")
                          }
                        )
  
  return(correlations)
}



# estimate the **significance** (p-value) of the pearson correlation of all genes compared to the selected gene


calc_pearson_sig <- function(dataset = all_q_norm, gene ="ESR1"){
  
  # get expression data for the 
  
  expression_data <- assayDataElement(dataset, elt = "q_norm")
  
  # get expression data for just the gene of interest
  target_expression <- assayDataElement(dataset, elt = "q_norm")[gene, ]
  
  correlations <- apply(expression_data, 1, 
                        function(gene_expression){
                          # function to compare correlation between gene and target selected
                          cor.test(gene_expression, target_expression, method = "pearson")$p.value
                          }
                        )
  
  return(correlations)
}
```



### FUNCT: corr_plot(), corr_plot_pairs()

plots correlation with pearson value and p-value for selected genes. 

```{r}

# this function plots correlation given a dataset and two genes to compare

corr_plot <- function(dataset = all_q_norm, gene_x = "ESR1", gene_y = "CXXC5", title_append = ""){
  
  # for whatever reason, x_vector and y_vector cannot be placed inside the aes() function as is and must be computed first. 
  # I have no idea why
  
  x_vector <- as.vector(assayDataElement(dataset, elt = "q_norm")[gene_x, ])
  y_vector <- as.vector(assayDataElement(dataset, elt = "q_norm")[gene_y, ])
  
  plot <- ggplot(
    pData(dataset), 
    aes(x = x_vector,
        y = y_vector
          )
      )+
  geom_point()+
  geom_smooth(formula = y ~ x, method = "lm")+ 
  stat_cor(method="pearson", size = 5)+ 
  labs(title= paste0(gene_x, " versus ", gene_y, title_append),
       x=gene_x, y = gene_y)+
  theme_classic()
  
  return(plot)
}

# this function create a paired graph of correlations for the discovery and validation datasets


corr_plot_pairs <- function(dataset1 = disc_q_norm, dataset2 = valid_q_norm, gene_x = "ESR1", gene_y = "CXXC5"){
  plot1 <- corr_plot(dataset = dataset1, gene_x = gene_x, gene_y = gene_y, title_append = ", Discovery")
  plot2 <- corr_plot(dataset = dataset2, gene_x = gene_x, gene_y = gene_y, title_append = ", Validation")
  
  plot_together <- plot1 + plot2
  
  return(plot_together)
  
  
}

```





### Create all OVGP1 / HLA Correlations


```{r}

fig5_genes <- c("HLA-DPA1", "HLA-DPB1", "HLA-DRA", "HLA-DRB", "HLA-DRB3", "HLA-DRB4", "HLA-DQA1", "HLA-DQB1", "HLA-DMA", "HLA-DMB", "HLA-DOA", "CD74", "ESR1", "PGR")

OVGP1_corr <- lapply(fig5_genes, compare_OVGP1, dataset = all_q_norm, title_append = "Expression v % OVGP1+ Cells") |>
  lapply(adj_text_size, pnt = 17)

names(OVGP1_corr) <- fig5_genes


```



### Create Figure 5

```{r fig.height=20, fig.width=20}

fig5_genes <- c("HLA-DPA1", "HLA-DPB1", "HLA-DRA", "HLA-DRB", "HLA-DRB3", "HLA-DRB4", 
                "HLA-DQA1", "HLA-DQB1", "HLA-DMA", "HLA-DMB", "HLA-DOA", "CD74", "C2", "C4B", "CFB")


make_fig5_boxplots <- function(genes = Fig5_genes, dataset = all_q_norm, title = "All"){

  graphs <- lapply(X = genes, FUN = Cil_v_Sec_Anat_Boxplot_Points, dataset = dataset)
  
  # resize
  graph_resize <- lapply(graphs, adj_text_size, pnt = 17)
  graph_remove_legend <- lapply(graph_resize, remove_legend)
  graph_sig <- lapply(graph_remove_legend, add_sig, pnt = 4)

  return(graph_sig)
  
  }


# Create all of the Boxplots for the Discovery Dataset

f5_disc <- make_fig5_boxplots(genes = fig5_genes , dataset = disc_q_norm)
names(f5_disc) <- fig5_genes

# Create all of the Boxplots for the Validation Dataset

f5_valid <- make_fig5_boxplots(genes = fig5_genes , dataset = valid_q_norm)
names(f5_valid) <- fig5_genes


# Create Grobs for use as labels

top <-   grobTree(rectGrob(gp=gpar(fill='#F0F0F0',col= 'black')), textGrob('Discovery Dataset: MHC-II transcripts', gp=gpar(fontsize=25)))

bottom <- grobTree(rectGrob(gp=gpar(fill='#F0F0F0',col= 'black')), textGrob('Validation Dataset: MHC-II transcripts', gp=gpar(fontsize=25)))

# create legends

legend_disc <- make_boxplot_legend(dataset = disc_q_norm)

legend_valid <- make_boxplot_legend(dataset = valid_q_norm)

# merge guides on OVGP1 comparison plots

OVGP1_plots <- remove_legend(OVGP1_corr$`HLA-DRA`) | remove_legend(OVGP1_corr$`HLA-DPA1`) | OVGP1_corr$`HLA-DQA1`

  # Part 1 - disc MHCII Transcripts
  wrap_elements(top) / 
  (f5_disc$`HLA-DRA` | f5_disc$`HLA-DRB` |  f5_disc$`HLA-DPA1` | f5_disc$`HLA-DQA1`) /
  legend_disc /
  # Part 2 - valid MHCII Transcripts  
  wrap_elements(bottom) / 
  (f5_valid$`HLA-DRA` | f5_valid$`HLA-DRB` |  f5_valid$`HLA-DPA1` | f5_valid$`HLA-DQA1`) /
  legend_valid /
  # Part 3 OVGP1 plots
  OVGP1_plots+
    
    plot_layout(heights = c(0.5, 4, 0.5, # Part 1
                            0.5, 4, 0.5, # Part 2
                            4))          # Part 3
    
    # plot_annotation(tag_levels = list(c("", "a", "b", "c", "d", 
    #                                     "", "", "e", "f", "g", "h", 
    #                                     "", "","i", "j", "k", "l", 
    #                                     "", "", "m", "n", "o", "p", "")), 
    #                                theme = theme(plot.tag = element_text(size = 30, face = "bold")))


```
